app/controllers/bookmarks_controller.rb

```diff
@@ -0,0 +1,64 @@
+class BookmarksController < ApplicationController
+  before_action :set_bookmark, only: [:show, :update, :destroy]
+  before_action :authenticate_user
+
+  # GET /bookmarks
+  def index
+    @bookmarks = Bookmark.all
+
+    render json: @bookmarks
+  end
+
+  # GET /bookmarks/1
+  def show
+    render json: @bookmark
+  end
+
+  # POST /bookmarks
+  def create
+    @bookmark = Bookmark.new(bookmark_params)
+
+    if @bookmark.save
+      render json: @bookmark, status: :created, location: @bookmark
+    else
+      render json: @bookmark.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /bookmarks/1
+  def update
+    if @bookmark.update(bookmark_params)
+      render json: @bookmark
+    else
+      render json: @bookmark.errors, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /bookmarks/1
+  def destroy
+    @bookmark.destroy
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_bookmark
+      @bookmark = Bookmark.find(params[:id])
+    end
+
+    # Only allow a list of trusted parameters through.
+    def bookmark_params
+      params.require(:bookmark).permit(:title, :url)
+    end
+
+    def authenticate_user
```

Comment:
This method should be part of application controller since it could be used by other controllers.

Reply:
Added to application_controller.rb